lm75bd/lm75bd.c

```
@@ -25,15 +27,42 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
   return ERR_CODE_SUCCESS;
 }
 
-error_code_t readTempLM75BD(uint8_t devAddr, float *temp) {
+error_code_t readTempLM75BD(uint8_t devAddr, float *temp)
+{
   /* Implement this driver function */
-  
+
+  /* write pointer register =0x00, read 2 bytes from sensor, convert 11-bit 2's complement temp format into celcius */
+
+  if (temp == NULL)
+    return ERR_CODE_INVALID_ARG;
+
+  error_code_t err;
+
+  uint8_t pointer = 0x00;
+  err = i2cSendTo(devAddr, &pointer, 1);
+  if (err != ERR_CODE_SUCCESS)
+    return err;
```

We not only want to return the error code from this function, but we want to log it if possible. We have a built-in macro that allows you to log the error code and return it.